packages/valory/skills/transaction_settlement_abci/rounds.py

```diff
@@ -472,6 +464,7 @@ class SynchronizeLateMessagesRound(CollectNonEmptyUntilThresholdRound):
     none_event = Event.NONE
     selection_key = "participant"
     collection_key = "late_arriving_tx_hashes"
+    _hash_length = TX_HASH_LENGTH
```

Comment:
`CollectNonEmptyUntilThresholdRound` returns a `list` so this is not going to work as expected here. In essence, we retrieve a list from each agent that has late-arriving hashes. This list contains strs with concatenated hashes.

Reply:
The incoming payload might be a concatenated set of tx_hashes, but it's not a list.
https://github.com/valory-xyz/open-autonomy/blob/c2bf8beda36c4ed5744a5cb2944aa357170919c3/packages/valory/skills/transaction_settlement_abci/payloads.py#L172-L175

Reply:
Please take a look at the `CollectNonEmptyUntilThresholdRound`, I believe it would help you understand my point. Also, notice that the `collection_key` has nothing to do with the `tx_hashes` property that you referenced.